jwst/stpipe/suffix.py

```diff
@@ -0,0 +1,386 @@
+"""Suffix manipulation
+
+API
+---
+`remove_suffix`
+    Main function used to remove suffixes from
+    file base names.
+
+Notes
+-----
+`KNOW_SUFFIXES` is the list used by `remove_suffix`. This
+list is generated by the function `find_suffixes`. `find_suffixes`
+searches the `jwst` codebase for possible suffixes. The function uses
+`SUFFIXES_TO_ADD` to add other suffixes that it would otherwise not
+discover or should absolutely be in the list. The function uses
+'SUFFIXES_TO_DISCARD` for strings found that are not to be considered
+suffixes.
+
+Hence, to update `KNOW_SUFFIXES`, update both `SUFFIXES_TO_ADD` and
+`SUFFIXES_TO_DISCARD` as necessary, then use the output of
+`find_suffixes`.
+"""
+from copy import copy
+from importlib import import_module
+from inspect import (getmembers, isclass)
+import logging
+from os import (listdir, path, walk)
+import re
+import sys
+
+__all__ = ['remove_suffix']
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+# Suffixes that are hard-coded or otherwise
+# have to exist. Used by `find_suffixes` to
+# add to the result it produces.
+SUFFIXES_TO_ADD = [
+    'cal', 'calints', 'crf', 'crfints',
```

Comment:
I don't see the standard AMI products in the list, which include `ami`, `amiavg`, and `aminorm`.